code/game/objects/items/stacks/money.dm

```diff
@@ -212,6 +239,40 @@
 		value = 4
 		return ..()
 
+/obj/item/stack/money/dollar100
+	name = "100 Dollar Bill"
+	desc = "Paper bank note valued at one-hundred dollars"
+	icon_state = "100dollar"
+	value = 100
+	amount = 1
+
+/obj/item/stack/money/dollar100/New()
+	if(amount < 50)
+		icon_state = "100dollar"
+	if(amount >= 50)
+		icon_state = "100dollar50"
+	if(amount >= 100)
+		icon_state = "100dollar100"
+	if(amount >= 300)
+		icon_state = "100dollar300"
+	if(amount == 500)
+		icon_state = "100dollar500"
+	update_icon()
```

Comment:
In the case of the New() proc, calling ..() to invoke the parent's New() constructor makes sense because you want to include the functionality defined in the parent's New() constructor, such as any initialization or setup that may be necessary for the stack object.